### PR TITLE
Changed Newsletter Enabled/Disable config settings by priority, store scope then default scope

### LINKED ISSUE #31188: Newsletter enabled setting is always retrieved from the default scope
When disabling the newsletter for store scope and website scope the newsletter checkbox is still visible in register form and the newsletter info-box is visible in customer account dashboard.

The newsletter enabled setting is always retrieved from the default scope

### Preconditions (*)

1. Magento Version: 2.3.5-p2

### Steps to reproduce (*)

1. Enable the newsletter for default scope in Stores > Settings > Configuration -> Customers -> Newsletter > General Options > Enabled
2. Disable newsletter for store scope.
3. Navigate to customer registration page

### Expected result (*)

1. the newsletter should be disabled in frontend. The newsletter subscription checkbox should not be visible in register form and the newsletter info-box should not be visible in customer account dashboard

### Actual result (*)

1. The newsletter subscription checkbox is visible in register form and the newsletter info-box is visible in customer account dashboard
Kuvakaappaus - 2020-12-07 13-50-58
Kuvakaappaus - 2020-12-07 13-49-55

\Magento\Newsletter\Model\Config::isActive
```
    /**
     * Returns newsletter's enabled status
     *
     * @param string $scopeType
     * @return bool
     */
    public function isActive(string $scopeType = ScopeConfigInterface::SCOPE_TYPE_DEFAULT): bool
    {
        return $this->scopeConfig->isSetFlag(self::XML_PATH_NEWSLETTER_ACTIVE, $scopeType);
    }

```

\Magento\Customer\Block\Form\Register::isNewsletterEnabled
```
    /**
     * Newsletter module availability
     *
     * @return bool
     */
    public function isNewsletterEnabled()
    {
        return $this->_moduleManager->isOutputEnabled('Magento_Newsletter')
            && $this->newsLetterConfig->isActive();
    }

```

### PR DESCRIPTION
**Pull request by saphaljha**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#31188

### Manual testing scenarios (*)

1. manually Testes

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by saphaljha** [comment]
@magento run all tests

File: app/code/Magento/Newsletter/Model/Config.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ public function __construct(
      * @param string $scopeType
      * @return bool
      */
-    public function isActive(string $scopeType = ScopeConfigInterface::SCOPE_TYPE_DEFAULT): bool
+    public function isActive(string $scopeType = \Magento\Store\Model\ScopeInterface::SCOPE_STORE): bool
```
Comment (gabrieldagama) [inline app/code/Magento/Newsletter/Model/Config.php]: I believe we shouldn't change the scope here, as you can see the scope is a parameter and can be requested differently by who is calling it, I would recommend changing `\Magento\Customer\Block\Form\Register::isNewsletterEnabled` to request the store scope instead.

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi, @saphaljha thanks for contributing! Please have a look at my comment below, also, we must cover the changes in this PR with automated tests in order to proceed with it.

Thanks!

File: app/code/Magento/Newsletter/Model/Config.php (hunk 1 above)
Comment (saphaljha) [reply app/code/Magento/Newsletter/Model/Config.php]: ok, I'll work on it

**Comment by engcom-Hotel** [comment]
Hello @saphaljha,
I will help with the fix and its test coverage.

**Comment by saphaljha** [comment]
@engcom-Hotel, ok, thank you!!

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by saphaljha** [comment]
Hello @gabrieldagama ,

Could you please review the pr?

Thanks you!

File: app/code/Magento/Newsletter/Test/Mftf/Test/StorefrontNewsletterSubscriptionDisabledForStoreViewAtRegistrationTest.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            <title value="Disabled Newsletter Subscription for store View"/>
+            <description value="Option to subscribe should not be displayed at registration form if it is switched off for current store"/>
+            <severity value="AVERAGE"/>
+            <group value="newsletter"/>
+            <group value="configuration"/>
+            <testCaseId value="MC-*"/>
+        </annotations>
+        <before>
+            <magentoCLI command="config:set --scope=stores --scope-code=default newsletter/general/active 0" stepKey="disableSubscriptionForStore"/>
+            <magentoCLI command="cache:clean config" stepKey="cleanCache"/>
```
Comment (gabrieldagama) [inline app/code/Magento/Newsletter/Test/Mftf/Test/StorefrontNewsletterSubscriptionDisabledForStoreViewAtRegistrationTest.xml]: Is clearing cache really required here?

File: app/code/Magento/Newsletter/Test/Mftf/Test/StorefrontNewsletterSubscriptionDisabledForStoreViewAtRegistrationTest.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        </before>
+        <after>
+            <magentoCLI command="config:set --scope=stores --scope-code=default newsletter/general/active 1" stepKey="enableSubscriptionForStore"/>
+            <magentoCLI command="cache:clean config" stepKey="cleanCacheAgain"/>
```
Comment (gabrieldagama) [inline app/code/Magento/Newsletter/Test/Mftf/Test/StorefrontNewsletterSubscriptionDisabledForStoreViewAtRegistrationTest.xml]: Is clearing cache really required here?

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @engcom-Hotel, thanks for adding test coverage here, can you please check my comments?

File: app/code/Magento/Newsletter/Test/Mftf/Test/StorefrontNewsletterSubscriptionDisabledForStoreViewAtRegistrationTest.xml (hunk 2 above)
Comment (engcom-Hotel) [reply app/code/Magento/Newsletter/Test/Mftf/Test/StorefrontNewsletterSubscriptionDisabledForStoreViewAtRegistrationTest.xml]: Has been removed.

File: app/code/Magento/Newsletter/Test/Mftf/Test/StorefrontNewsletterSubscriptionDisabledForStoreViewAtRegistrationTest.xml (hunk 3 above)
Comment (engcom-Hotel) [reply app/code/Magento/Newsletter/Test/Mftf/Test/StorefrontNewsletterSubscriptionDisabledForStoreViewAtRegistrationTest.xml]: Has been removed.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Review by gabrieldagama — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8675](https://jira.corp.magento.com/browse/ENGCOM-8675) has been created to process this Pull Request
:eight_spoked_asterisk: @gabrieldagama, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario
1. From Admin Panel, go to _Stores - Configuration - Customers - Newsletter_
2. Select **Scope** - _Default Config_ (should be selected by default)
3. Expand **General Options** and set **Yes** for _Enabled_ (should be selected by default)
4. **Save Config**
5. Change the  **Scope** to _Default Store View_
6. Expand **General Options** and set **No** for _Enabled_, **Save Config**
7. Go to Storefront, default store view and click **Create an Account**
8. Verify if the **Sign up for Newsletter** checkbox is available in the customer registration form

**BEFORE** applying changes provided in the PR, the **Sign up for Newsletter** checkbox was available despite that it has been disabled from Admin Panel
sign_up_before

**AFTER** switching to the PR, the **Sign up for Newsletter** checkbox is not available
sign_up_after

- tested in Scope of multiple storeviews
- tested a case when a Customer is registered when the _Newsletter subscription_ is Disabled, and then changed the configuration and Enabled the _Newsletter subscription_. Customer can successfully Subscribe for Newsletter
subscr_success

**Pull request merged**